### ISSUE TITLE
Add support for URL/URI nodes

### ISSUE THREAD

**Issue by noschang** [lukas-krecan/JsonUnit#232]
I need to test a JSON which contains URLS on its nodes as the following example:
```json 
"_links": {
        "first": {
            "href": "http://localhost/api/users?page=0&size=3&sort=id,asc"
        }
}
```

It would be nice if I could test it with the following API:
```java
assertThatJson(result).node("_links.first.href").and(

        node -> node.asURL().protocolIs("http"),
        node -> node.asURL().domainIs("localhost"),
        node -> node.asURL().portIsAbsent(),
        node -> node.asURL().pathIs("/api/users"),
		
        node -> node.asURL().param("page").isEqualTo(0),
        node -> node.asURL().param("size").isEqualTo(3),
        node -> node.asURL().param("sort").isEqualTo("id,asc")
);
```
This follows the convention of your current API. But it would be better if I could convert the node to a URL before chaining the ```and()``` method, as follows:

```java
assertThatJson(result).node("_links.first.href").asURL().and(

        url -> url.protocolIs("http"),
        url -> url.domainIs("localhost"),
        url -> url.portIsAbsent(),
        url -> url.pathIs("/api/users"),

        url -> url.param("page").isEqualTo(0),
        url -> url.param("size").isEqualTo(3),
        url -> url.param("sort").isEqualTo("id,asc")
);
```

So, it's just an idea. I know I can do the same with something like ```node.asString().contains("page=0")```, but it doesn't seem as good

**Comment by lukas-krecan** [comment]
Hi, thanks for feedback. Seems legit, will think about it.

**Comment by lukas-krecan** [comment]
I have added support for URI assertions. It is possible to chain them like this

```java
assertThatJson("{\"a\":{\"b\":\"http://exampl.org?a=1\"}}")
            .node("a.b")
            .isUri().hasScheme("http").hasParameter("a", "1");
```

**Comment by lukas-krecan** [comment]
Released as 2.18.0
